src/components/VideoCard/VideoDetails.jsx

```diff
@@ -1,109 +1,137 @@
-import { useNavigate } from "react-router-dom";
+import { useNavigate, useLocation } from "react-router-dom";
 
 import { usePlaylist, useAuth } from "../../context/";
 import { isLiked, isInWatchlater } from "../../utils";
+import { Modal } from "../Modal/Modal";
+import { useModal } from "../../context";
 import { addToLikedVideo, removeFromLikedVideo, addToWatchLater, removeFromWatchLater } from "../../playlistServices";
 import "./VideoDetails.css"
 
 export const VideoDetails = ({_id, title, channelName, views, description, singleVideo}) => {
-    const { likedVideos, watchLater , playlistDispatch } = usePlaylist();
-    const liked = isLiked(likedVideos, _id);
-    const watchlater = isInWatchlater(watchLater, _id);
 
-    const {eToken} = useAuth();
+  const { likedVideos, watchLater , playlistDispatch } = usePlaylist();
+  const liked = isLiked(likedVideos, _id);
+  const watchlater = isInWatchlater(watchLater, _id);
 
-    const navigate = useNavigate();
+  const { modalDispatch, isModalOpen } = useModal();
 
-    const handleLikeClick = async () => {
-      if (eToken){
-        if (!liked){
-          const likes = await addToLikedVideo(singleVideo);
-          likes.map(video => playlistDispatch({
-            type: "LIKED",
-            payload: video
-          }))
-          
-        }else{
-          const removedLikes = await removeFromLikedVideo(singleVideo);
-          playlistDispatch({
-            type: "REMOVE_FROM_LIKE",
-            payload: singleVideo
-          })
-          
-        }
+  const {eToken} = useAuth();
+
+  const navigate = useNavigate();
+
+  const {pathname} = useLocation();
+
+  const handleLikeClick = async () => {
+    if (eToken){
+      if (!liked){
+        const likes = await addToLikedVideo(singleVideo);
+        likes && likes.map(video => playlistDispatch({
+          type: "LIKED",
+          payload: video
+        }))
+        
```

Comment:
can you explain what is happening over here

Reply:
so this I did beacuse, the api "/api/user/likes" returns an array of objects, but i want only one object to be added to liked videos, so that's why I did the above